src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearing/AssignSessionsToHearingAction.java

```diff
@@ -153,40 +152,18 @@ public void act() {
 
     @Override
     public List<UserTransactionData> generateUserTransactionData() {
-
-        List<UserTransactionData> userTransactionDataList = new ArrayList<>();
-        userTransactionDataList.add(new UserTransactionData("hearing",
-            savedHearing.getId(),
-            null,
-            "lock",
-            "unlock",
-            0)
-        );
-
-        userTransactionDataList.add(new UserTransactionData("hearing",
-            savedHearing.getId(),
-            previousHearing,
-            UPDATE_ACTION_TEXT,
-            UPDATE_ACTION_TEXT,
-            1)
+        originalHearingParts.forEach((id, hpString) ->
```

Comment:
just to confirm, was part of the story to revert the order of rollback?

Reply:
Just to check that user_transaction_data was being set properly.
Although it would make more sense to unlock the hearing part and session first before updating the hearingPart; so i've reverted it back.